### PR TITLE
(PC-6434): move SafeAreaProvider at the top of the app, remove naviga…

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
…tion context from AsyncErrorBoundaryWithoutNavigation

Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-XXX

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/errors/pages/AsyncErrorBoundary.tsx [hunk 1]
```diff
@@ -91,6 +86,23 @@ export const AsyncErrorBoundary = ({
   )
 }
 
+export const AsyncErrorBoundary = ({ backNavigation = true, ...rest }: AsyncFallbackProps) => {
```
Comment (antoinewg) [inline src/features/errors/pages/AsyncErrorBoundary.tsx:89]: je propose de changer ce composant en ça:

```js
export const AsyncErrorBoundary = (props: AsyncFallbackProps) => {
  const { canGoBack, goBack } = useNavigation<UseNavigationType>()
  const { top } = useCustomSafeInsets()

  return (
    <AsyncErrorBoundaryWithoutNavigation
      {...props}
      header={() =>
        canGoBack() && (
          <HeaderContainer onPress={goBack} top={top + getSpacing(3.5)} testID="backArrow">
            <ArrowPrevious color={ColorsEnum.WHITE} size={getSpacing(10)} />
          </HeaderContainer>
        )
      }
    />
  )
}
```

donc plusieurs trucs:
 - retrait du backNavigation.
 - navigation => header => passé directement
 - pas de useMemo. tu peux le rajouter ça me dérange pas.
 - pas de "destructuration" de `props` en `rest`.

**Review by antoinewg — APPROVED** [review]
Mis à part certains tests à modifier / supprimer, on arrive à la même solution :)
J'ai testé ça marche bien, merci 👍

File: src/features/errors/pages/AsyncErrorBoundary.tsx (hunk 1 above)
Comment (kopax) [reply src/features/errors/pages/AsyncErrorBoundary.tsx:89]: Une raison pour choisir un `ReactComponent` plutot que un `ReactElement` pour la prop `header` ?

Comment (antoinewg) [reply src/features/errors/pages/AsyncErrorBoundary.tsx]: ça revient au même, je sais pas lequel est le plus fancy. Aucune raison particulière. Tu peux merger

**Pull request merged**